cypress/support/step_definitions/testSetup.js

```diff
@@ -55,6 +55,21 @@ Given('the environment has been set up for {string} tests', (test) => {
     if (Cypress.env(CONSTANTS.TEST_TYPE).includes('rpc-Only')) {
       Cypress.env(CONSTANTS.IS_RPC_ONLY, true);
     }
+    // fetch device details dynamically
+    if (CONSTANTS.DYNAMIC_DEVICE_DETAILS_MODULES.includes(Cypress.env(CONSTANTS.TEST_TYPE))) {
+      cy.getDeviceData(CONSTANTS.DEVICE_ID, {}, CONSTANTS.ACTION_CORE.toLowerCase()).then(
+        (response) => {
+          if (response) {
+            const method = CONSTANTS.REQUEST_OVERRIDE_CALLS.FETCHDEVICEDETAILS;
+            const requestMap = {
+              method: method,
+              params: response,
+            };
+            cy.sendMessagetoPlatforms(requestMap);
+          }
+        }
+      );
+    }
```

Comment:
@kevinshahfws  , I feel we should remove the dependency of fetching device from FCS . Instead we can just have something like below

// If testtype == module which is included in DYNAMIC_DEVICE_DETAILS_MODULES list in constant , fetch device details dynamically
`if (CONSTANTS.DYNAMIC_DEVICE_DETAILS_MODULES.includes(Cypress.env(CONSTANTS.TEST_TYPE))) {
       // creating a request map to fetch device details from config module. Config module should have fetchDeviceDetails function inside fcs.js defined which would generate source of truth and save data in env
            const method = CONSTANTS.REQUEST_OVERRIDE_CALLS.FETCHDEVICEDETAILS;
            const requestMap = {
              method: method,
              params: response,
            };
            cy.sendMessagetoPlatforms(requestMap);
          }
        }
      );
    }`